data_iterator.go

```diff
@@ -189,3 +204,62 @@ func (d *DataIterator) AddBatchListener(listener func(*RowBatch) error) {
 func (d *DataIterator) AddDoneListener(listener func() error) {
 	d.doneListeners = append(d.doneListeners, listener)
 }
+
+func GetOrderedTables(unorderedTables map[*TableSchema]uint64, db *sql.DB) ([]TableKeyInfo, error) {
+	orderedTables := make([]TableKeyInfo, len(unorderedTables))
+	i := 0
+
+	for k, v := range unorderedTables {
+		orderedTables[i] = TableKeyInfo{k, v}
+		i++
+	}
+
+	tableNames := make([]string, len(unorderedTables))
+	databaseSchemasMap := make(map[string]int, len(unorderedTables))
+
+	i = 0
+	for t := range unorderedTables {
+			tableNames[i] = t.Name
+			databaseSchemasMap[t.Schema] = 0
+			i++
+	}
+
+	databaseSchemas := make([]string, len(databaseSchemasMap))
+
+	i = 0
+	for schema := range databaseSchemasMap {
+		databaseSchemas[i] = schema
+	}
+
+	query := fmt.Sprintf("SELECT TABLE_NAME, TABLE_SCHEMA, DATA_LENGTH FROM information_schema.tables WHERE TABLE_SCHEMA IN (\"%s\") AND TABLE_NAME IN (\"%s\") ORDER BY DATA_LENGTH DESC", strings.Join(databaseSchemas, `", "`), strings.Join(tableNames, `", "`))
+	rows, err := db.Query(query)
```

Comment:
Alternatively, to make this code simpler (at the code of performance), you can just issue one of these SELECT queries per table in the list?